pkg/cephmgr/rgw/bucket.go

```diff
@@ -49,53 +82,136 @@ func GetBucketStats(context *clusterd.Context, getClusterInfo func() (*mon.Clust
+		buckets = append(buckets, model.ObjectBucket{Name: bucket, ObjectBucketMetadata: model.ObjectBucketMetadata{Owner: metadata.Owner, CreatedAt: metadata.CreatedAt}, ObjectBucketStats: stat})
+	}
+
+	return buckets, nil
+}
+
+func GetBucket(context *clusterd.Context, bucket string, getClusterInfo func() (*mon.ClusterInfo, error)) (*model.ObjectBucket, int, error) {
+	stat, err := GetBucketStats(context, bucket, getClusterInfo)
+	if err != nil {
+		if err.Error() == "not found" {
+			return nil, RGWErrorNotFound, fmt.Errorf("Bucket not found")
 		}
+		return nil, RGWErrorUnknown, fmt.Errorf("Failed to get bucket stats: %+v", err)
+	}
 
-		createdAt, err := time.Parse("2006-01-02 15:04:05.999999999Z", s.Data.CreationTime)
-		if err != nil {
-			return nil, fmt.Errorf("Error parsing date (%s): %+v", s.Data.CreationTime, err)
+	metadata, err := getBucketMetadata(context, bucket, getClusterInfo)
+	if err != nil {
+		if err.Error() == "not found" {
+			return nil, RGWErrorNotFound, fmt.Errorf("Bucket not found")
 		}
+		return nil, RGWErrorUnknown, err
+	}
 
-		buckets = append(buckets, model.ObjectBucket{Name: bucket, Owner: s.Data.Owner, CreatedAt: createdAt, ObjectBucketStats: stat})
+	return &model.ObjectBucket{Name: bucket, ObjectBucketMetadata: model.ObjectBucketMetadata{Owner: metadata.Owner, CreatedAt: metadata.CreatedAt}, ObjectBucketStats: *stat}, RGWErrorNone, nil
+}
+
+// FIXME: The "bucket create" radowsgw-admin command does not exist yet
+func CreateBucket(context *clusterd.Context, bucket model.ObjectBucket, getClusterInfo func() (*mon.ClusterInfo, error)) (*model.ObjectBucket, error) {
+	/* Proposed implementation
+	result, err := RunAdminCommand(context, getClusterInfo,
+		"bucket",
+		"create",
+		"--uid", bucket.Owner,
+		"--bucket", bucket.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create bucket: %+v", err)
 	}
 
-	return buckets, nil
+	if result != "bucket created" {
+		return nil, fmt.Errorf("failed to create bucket: %s", result)
+	}
+
+	return GetBucket(context, bucket.Name, getClusterInfo)
+	*/
+
+	return nil, fmt.Errorf("Not yet implemented")
+}
+
+func DeleteBucket(context *clusterd.Context, bucketName string, purge bool, getClusterInfo func() (*mon.ClusterInfo, error)) error {
+	options := []string{"--bucket", bucketName}
+	if purge {
+		options = append(options, "--purge-objects")
+	}
+
+	result, err := RunAdminCommand(context, getClusterInfo,
+		"bucket",
+		"rm",
+		options...)
+	if err != nil {
+		return fmt.Errorf("failed to delete bucket: %+v", err)
+	}
+
+	if result == "" {
+		return nil
+	}
+
+	if strings.Contains(result, "could not get bucket info for bucket=") {
+		return fmt.Errorf("bucket not found")
+	}
+
+	return fmt.Errorf("failed to delete bucket: %+v", err)
+}
+
+func GetBucketACL(context *clusterd.Context, bucketName string, getClusterInfo func() (*mon.ClusterInfo, error)) (*model.ObjectBucketACL, error) {
+	return nil, fmt.Errorf("Not yet implemented")
```

Comment:
i think we either need to remove the ACL placeholders in this PR, or finish the implementation before the PR is merged